src/legacy/core_plugins/interpreter/public/functions/tagcloud.js

```diff
@@ -31,13 +31,63 @@ export const tagcloud = () => ({
     defaultMessage: 'Tagcloud visualization'
   }),
   args: {
-    visConfig: {
-      types: ['string', 'null'],
-      default: '"{}"',
+    scale: {
+      types: ['string'],
```

Comment:
should we (and how can we) provide allowed options here ? ('linear', 'log', 'sqrt')
we already have this check on the chart level

Reply:
@w33ble ^^ whats the canvas way ?

Reply:
Use the `options` argument for the function definition, since the auto-complete dialog will show those values. Then in the function itself, just throw if the value is invalid. An example of this is [alterColumn](https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/canvas_plugin_src/functions/common/alterColumn.js#L62-L78)